framework/src/node/consensus/certificate_generation/commit_pool.ts

```diff
@@ -57,13 +58,25 @@ export class CommitPool {
 	}
 
 	public createSingleCommit(
-		_blockHeader: BlockHeader,
-		_validatorInfo: ValidatorInfo,
-		_networkIdentifier: Buffer,
-		// eslint-disable-next-line @typescript-eslint/no-empty-function
+		blockHeader: BlockHeader,
+		validatorInfo: ValidatorInfo,
+		networkIdentifier: Buffer,
 	): SingleCommit {
-		return {} as SingleCommit;
+		const commit: SingleCommit = {
+			blockID: blockHeader.id,
+			height: blockHeader.height,
+			validatorAddress: validatorInfo.address,
+			certificateSignature: undefined,
+		};
+		commit.certificateSignature = signCertificate(
```

Comment:
instead of assigning to the `commit` object here, if signature is calculated first, the certificate type doesn't need to be changed. ex)
```
const certificateSignature = signCertificate(xxx);
return {
  blockID: blockHeader.id,
  height: blockHeader.height,
  validatorAddress: validatorInfo.address,
  certificateSignature,
};
```